dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -1981,35 +1912,223 @@ public Contentlet save(Contentlet contentlet) throws DotDataException, DotStateE
 	}
 
 	@Override
-	protected Contentlet save(Contentlet contentlet, String existingInode) throws DotDataException, DotStateException, DotSecurityException {
-	    com.dotmarketing.portlets.contentlet.business.Contentlet fatty = new com.dotmarketing.portlets.contentlet.business.Contentlet();
-        if(InodeUtils.isSet(contentlet.getInode())){
-            fatty = (com.dotmarketing.portlets.contentlet.business.Contentlet)HibernateUtil.load(com.dotmarketing.portlets.contentlet.business.Contentlet.class, contentlet.getInode());
+    protected Contentlet save(Contentlet contentlet, String existingInode)
+            throws DotDataException, DotStateException, DotSecurityException {
+
+        final String inode = getInode(existingInode, contentlet);
+        upsertContentlet(contentlet, inode);
+        contentlet.setInode(inode);
+
+
+        contentlet.getMap().remove(WORKFLOW_ACTION_KEY);
```

Comment:
I would group this in to a removableSet and uses such as 
removableSet.forEach(key -> content.getMap().remove(key));

Reply:
it would be a class attribute (constant) 
not sure if a configurable, think about it

Reply:
Done